pkg/agent/agent.go

```diff
@@ -496,27 +487,77 @@ func (c *agentController) CreateAddOnPlacementScore() {
 			Name:      util.HostedClusterScoresResourceName,
 			Namespace: c.clusterName,
 		},
-		Status: clusterv1alpha1.AddOnPlacementScoreStatus{
-			Scores: items,
-		},
 	}
 
-	_, err = controllerutil.CreateOrUpdate(context.TODO(), c.hubClient, addOnPlacementScore, func() error { return nil })
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), c.hubClient, addOnPlacementScore, func() error { return nil })
 	if err != nil {
 		// just log the error. it should not stop the rest of reconcile
 		c.log.Error(err, fmt.Sprintf("failed to create or update the addOnPlacementScore resource in %s", c.clusterName))
-		return
+		return err
 	}
 
-	addOnPlacementScore.Status.Scores = items
-	err = c.hubClient.Status().Update(context.TODO(), addOnPlacementScore, &client.UpdateOptions{})
+	listopts := &client.ListOptions{}
+	hcList := &hyperv1alpha1.HostedClusterList{}
+	err = c.spokeUncachedClient.List(context.TODO(), hcList, listopts)
 	if err != nil {
 		// just log the error. it should not stop the rest of reconcile
-		c.log.Error(err, fmt.Sprintf("failed to update the addOnPlacementScore status in %s", c.clusterName))
-		return
+		c.log.Error(err, "failed to get HostedCluster list")
+
+		conditions := []metav1.Condition{
+			{
+				LastTransitionTime: metav1.Time{Time: time.Now()},
+				Type:               "HostedClusterCountUpdated",
+				Status:             metav1.ConditionFalse,
+				Reason:             "HostedClusterCountFailed",
+				Message:            err.Error(),
+			},
+		}
+
+		// Just add conditions, no score or keep the existing score
+		addOnPlacementScore.Status.Conditions = conditions
+
+		err = c.hubClient.Status().Update(context.TODO(), addOnPlacementScore, &client.UpdateOptions{})
+		if err != nil {
+			// just log the error. it should not stop the rest of reconcile
+			c.log.Error(err, fmt.Sprintf("failed to update the addOnPlacementScore status in %s", c.clusterName))
+			return err
+		}
+	} else {
+		scores := []clusterv1alpha1.AddOnPlacementScoreItem{
+			{
+				Name:  util.HostedClusterScoresScoreName,
+				Value: int32(len(hcList.Items)),
```

Comment:
Is it possible to run into a nilpointer exception here? Not sure if it's possible that `hcList` becomes nil after a `List` call with no err.

Reply:
I don't think that can happen...

Reply:
As long as CRD exists, it will return an empty list or something. Otherwise, an error will be caught.